### ISSUE TITLE
Unspecific Strelka issue with WGS Tumor-Normal Pair

### ISSUE THREAD

**Issue by Alfredo-Enrique** [uclahs-cds/pipeline-call-sSNV#262]
**Describe the bug**
Running call-sSNV on a tumor-normal Whole-Genome Sequencing Pair and getting this error below. I narrowed it down just to strelka and performed a call-sSNV launch with only strelka to ensure. I think it might have something to do with a cigar string? Unsure if it's a memory problem but doubt it.

* Pipeline release version : 8.0.0-rc.1
* Cluster you are using : Slurm
* Node type : F72
* Submission method : sbatch script  (Not python submission package), maybe my scratch directories are wrong? I don't think so, based on the log it seems to go through and just gets hung up on a random chromosome or cigar string. Same launch method works for the other processes so it's strelka specific.
* Actual submission script (python submission script, "nextflow run ...", etc.)
From `/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/run_script/launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017.sh`
```
#!/bin/bash
launch_time=$(date +"%Y.%m.%d-%H:%M")
log=/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/run_script/log 
sbatch --exclusive --partition=F72 \
-J run_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017-T002-P02-F_strelka_$launch_time \
-e $log/run_launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017-T002-P02-F_strelka_$launch_time.error \
-o $log/run_launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017-T002-P02-F_strelka_$launch_time.log \
--wrap="TEMP_DIR=\$(mktemp -d /scratch/\$SLURM_JOB_ID/XXXXXXX) && cd \$TEMP_DIR && nextflow run /hot/user/alfgonzalez/pipeline/pipeline-call-sSNV/8.0.0-rc.1/main.nf \
        -config /hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/config/template.config \
        -params-file '/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/input/SGSALASC000017-T002-P02-F.yaml'" \
--mail-user=alfgonzalez@mednet.ucla.edu --mail-type=ALL
```

* Sbatch or qsub command and logs if applicable : Above and below
* Config files: above
* Path to the working directory : `/hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/.command.log`
/hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/
* Any logs produced by the pipeline:
sbatch log
`/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/run_script/log/run_launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017-T002-P02-F_strelka_2023.12.28-15:08.log`
process specific log
`/hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/.command.log `

**To Reproduce**
1. Adjust config file as needed for working directory and output location `/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/config/template.config`
2. Adjust desired log location on run script `/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/run_script/launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017.sh`
3. Launch runs script `/hot/user/alfgonzalez/project/project-LAND-SARC/project-SARC-LAND/pipeline/SGSALASC/call-sSNV/run_script/launch_pipeline-call-sSNV-8.0.0-rc.1_SGSALASC000017.sh`

**Expected behavior**
It will fail and produce the following error below. Note that this is only the final excerpt once the error comes up. 

```
[2023-12-29T04:23:35.042196Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] ===== StrelkaSomaticWorkflow StatusUpdate =====
[2023-12-29T04:23:35.043064Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Workflow specification is complete?: True
[2023-12-29T04:23:35.043088Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Task status (waiting/queued/running/complete/error): 760/0/1/776/1
[2023-12-29T04:23:35.043106Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Longest ongoing queued task time (hrs): 0.0000
[2023-12-29T04:23:35.043120Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Longest ongoing queued task name: ''
[2023-12-29T04:23:35.043133Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Longest ongoing running task time (hrs): 1.4686
[2023-12-29T04:23:35.043145Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [StatusUpdate] Longest ongoing running task name: 'CallGenome+callGenomeSegment_chromId_045_chr14_GL000225v1_random_0000'
[2023-12-29T04:27:53.977530Z] [977c2e4e3c8e] [91_1] [TaskManager] Completed command task: 'CallGenome+callGenomeSegment_chromId_045_chr14_GL000225v1_random_0000' launched from sub-workflow 'CallGenome'
[2023-12-29T04:28:01.873346Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] Workflow terminated due to the following task errors:
[2023-12-29T04:28:01.874261Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] Failed to complete command task: 'CallGenome+callGenomeSegment_chromId_004_chr5_0004' launched from sub-workflow 'CallGenome', error code: 1, command: '/usr/local/share/strelka-2.9.10-1/libexec/strelka2 --region chr5:45384568-56730708 --ref /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/Homo_sapiens_assembly38.fasta --max-indel-size 49 --candidate-indel-input-vcf /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/candidateSmallIndels.vcf.gz --call-regions-bed /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/Homo_sapiens_assembly38_no-decoy.bed.gz --min-mapping-quality 20 --somatic-snv-rate 0.0001 --shared-site-error-rate 0.0000000005 --shared-site-error-strand-bias-fraction 0.0 --somatic-indel-rate 0.000001 --shared-indel-error-factor 2.2 --tier2-min-mapping-quality 0 --strelka-snv-max-filtered-basecall-frac 0.4 --strelka-snv-max-spanning-deletion-frac 0.75 --strelka-snv-min-qss-ref 15 --strelka-indel-max-window-filtered-basecall-frac 0.3 --strelka-indel-min-qsi-ref 40 --ssnv-contam-tolerance 0.15 --indel-contam-tolerance 0.15 --somatic-snv-scoring-model-file /usr/local/share/strelka-2.9.10-1/share/config/somaticSNVScoringModels.json --somatic-indel-scoring-model-file /usr/local/share/strelka-2.9.10-1/share/config/somaticIndelScoringModels.json --normal-align-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/BWA-MEM2-2.2.1_GATK-4.2.4.1_SARC-000118-SarcomaLandscape_SGSALASC000017-N001-B01-F.bam --tumor-align-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/BWA-MEM2-2.2.1_GATK-4.2.4.1_SARC-000118-SarcomaLandscape_SGSALASC000017-T002-P02-F.bam --somatic-snv-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/StrelkaSomaticWorkflow/workspace/genomeSegment.tmpdir/somatic.snvs.unfiltered.chromId_004_chr5_0004.vcf --somatic-indel-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/StrelkaSomaticWorkflow/workspace/genomeSegment.tmpdir/somatic.indels.unfiltered.chromId_004_chr5_0004.vcf --stats-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/StrelkaSomaticWorkflow/workspace/genomeSegment.tmpdir/runStats.chromId_004_chr5_0004.xml --strelka-skip-header --strelka-chrom-depth-file /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/StrelkaSomaticWorkflow/workspace/chromDepth.tsv --strelka-max-depth-factor 3.0'
[2023-12-29T04:28:01.874313Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] Error Message:
[2023-12-29T04:28:01.874331Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] Anomalous task wrapper stderr output. Wrapper signal file: '/hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.0/workdir/09/59dc135e0977893f32e1958d4ad15d/StrelkaSomaticWorkflow/workspace/pyflow.data/logs/tmp/taskWrapperLogs/000/510/pyflowTaskWrapper.signal.txt'
[2023-12-29T04:28:01.874356Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] Logging 11 line(s) of task wrapper log output below:
[2023-12-29T04:28:01.874376Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:46:56.973555Z] [977c2e4e3c8e] [91_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_004_chr5_0004] [wrapperSignal] wrapperStart
[2023-12-29T04:28:01.874405Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:46:57.002121Z] [977c2e4e3c8e] [91_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_004_chr5_0004] [wrapperSignal] taskStart
[2023-12-29T04:28:01.874427Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:51:10.984518Z] [977c2e4e3c8e] [91_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_004_chr5_0004] [wrapperSignal] taskExitCode -9
[2023-12-29T04:28:01.874453Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:51:14.420306Z] [977c2e4e3c8e] [91_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_004_chr5_0004] [wrapperSignal] taskStderrTail 7
[2023-12-29T04:28:01.874478Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] Last 6 stderr lines from task (of 6 total lines):
[2023-12-29T04:28:01.874504Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:46:57.033225Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: theta parameter file was not given. Using internal theta values.
[2023-12-29T04:28:01.874529Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:47:20.141180Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: skipping alignment for read outside of buffer range: A00454:355:H3L3KDSXY:1:2151:18710:6778/2 tid:pos:strand 4:46164782:- cigar: 5M1I145M templSize: -209 mate_tid:pos:strand 4:46164723:+
[2023-12-29T04:28:01.874560Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:48:49.331190Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: skipping alignment for read outside of buffer range: A00454:355:H3L3KDSXY:2:2561:9001:6903/1 tid:pos:strand 4:49658652:+ cigar: 5I125M21S templSize: 104 mate_tid:pos:strand 4:49658657:-
[2023-12-29T04:28:01.874615Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:48:49.340646Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: skipping alignment for read outside of buffer range: A00454:355:H3L3KDSXY:4:1246:28845:1783/2 tid:pos:strand 4:49658652:+ cigar: 5I118M28S templSize: -57136 sa: chrUn_KN707896v1_decoy,7063,+,11S53M1I86M,1,10; mate_tid:pos:strand 4:49601517:+
[2023-12-29T04:28:01.874661Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:48:49.360194Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: skipping alignment for read outside of buffer range: A00454:355:H3L3KDSXY:2:1621:19045:8860/1 tid:pos:strand 4:49658653:+ cigar: 5I134M12S templSize: 2509 mate_tid:pos:strand 4:49661040:-
[2023-12-29T04:28:01.874688Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] [taskWrapper-stderr] [2023-12-29T03:48:49.369158Z] [977c2e4e3c8e] [91_1] [CallGenome+callGenomeSegment_chromId_004_chr5_0004] WARNING: skipping alignment for read outside of buffer range: A00454:355:H3L3KDSXY:3:1246:31964:35556/2 tid:pos:strand 4:49658653:+ cigar: 5I118M27S templSize: 0 sa: chrUn_KN707896v1_decoy,373,+,5I107M5D31M7S,0,21; mate_tid:pos:strand 1:89828621:-
[2023-12-29T04:28:01.874726Z] [977c2e4e3c8e] [91_1] [WorkflowRunner] [ERROR] Failed to complete sub-workflow task: 'CallGenome' launched from master workflow, failed sub-workflow classname: 'CallWorkflow'

```
Thank-you in advance for any insights or suggestions you may have! Unsure if anyone has seen a similar error before or if it's something trivial like increasing memory allocation.

**Comment by sorelfitzgibbon** [comment]
@Alfredo-Enrique I have not seen this before.  Looks like it failed in different places for the original and `strelka2` only runs.   As best I can tell, I think the relevant error is `Anomalous task wrapper stderr output` and can be caused by a [linux out-of-memory kill](https://github.com/Illumina/strelka/issues/37#:~:text=%40akm0001%20%2C%20I%20think,it%20isn%27t%20local%3F).  

 Is the sample unusually large?

**Comment by Alfredo-Enrique** [comment]
Pretty sure it is 60x-90x coverage so not unusually large for a tumor file. The normal is also probably around 30x- to 40x.

125 Gb
- /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.1/SARC-000118-SarcomaLandscape/main_workflow/output/recalibrate-BAM-1.0.0-rc.4/SGSALASC000017-T002-P02-F/GATK-4.2.4.1/output/BWA-MEM2-2.2.1_GATK-4.2.4.1_SARC-000118-SarcomaLandscape_SGSALASC000017-N001-B01-F.bam
291 Gb
- /hot/project/disease/SarcomaTumor/SARC-000118-SarcomaLandscape/data/SGSALASC/WGS/metapipeline-DNA-5.2.1/SARC-000118-SarcomaLandscape/main_workflow/output/recalibrate-BAM-1.0.0-rc.4/SGSALASC000017-T002-P02-F/GATK-4.2.4.1/output/BWA-MEM2-2.2.1_GATK-4.2.4.1_SARC-000118-SarcomaLandscape_SGSALASC000017-T002-P02-F.bam

**Comment by sorelfitzgibbon** [comment]
OK thanks, I'll run some tests.

**Comment by Alfredo-Enrique** [comment]
Making a quick note in case it helps someone in the future! Following discussions with Sorel, it seems that when strelka2 is not given enough memory, the error it outputs does not trigger the nextflow-retry strategy. 

For the files above, I increased the memory substantially and the module/pipeline finished without any errors.
<img width="271" alt="image" src="https://github.com/uclahs-cds/pipeline-call-sSNV/assets/60890801/34578cf7-9b18-4233-b5bf-c602f2987f5c">

After we looked at the trace file, we did see that the utilized memory in the process was about 0.5Gb higher than allocated in the default settings. However, the default retry strategy was never triggered, and if it had, it would have been more then sufficient to cover the memory requirements.

**Comment by bltsai-UCLA** [comment]
@Alfredo-Enrique I'm also getting this error. Did you clone your own instance of the pipeline, then modify those parameters there?

**Comment by sorelfitzgibbon** [comment]
> @Alfredo-Enrique I'm also getting this error. Did you clone your own instance of the pipeline, then modify those parameters there?

@bltsai-UCLA you can increase the memory for the process using [base_resource_update](https://github.com/uclahs-cds/pipeline-call-sSNV/tree/main?tab=readme-ov-file#base-resource-allocation-updaters) in the config file.

**Comment by sorelfitzgibbon** [comment]
> > @Alfredo-Enrique I'm also getting this error. Did you clone your own instance of the pipeline, then modify those parameters there?
> 
> @bltsai-UCLA you can increase the memory for the process using [base_resource_update](https://github.com/uclahs-cds/pipeline-call-sSNV/tree/main?tab=readme-ov-file#base-resource-allocation-updaters) in the config file.

What version of call-sSNV are you using?

**Comment by bltsai-UCLA** [comment]
> > > @Alfredo-Enrique I'm also getting this error. Did you clone your own instance of the pipeline, then modify those parameters there?
> > 
> > 
> > @bltsai-UCLA you can increase the memory for the process using [base_resource_update](https://github.com/uclahs-cds/pipeline-call-sSNV/tree/main?tab=readme-ov-file#base-resource-allocation-updaters) in the config file.
> 
> What version of call-sSNV are you using?

This was from metapipeline-DNA v5.2.1. I'll try again with the most updated version.

**Comment by bltsai-UCLA** [comment]
> > > > @Alfredo-Enrique I'm also getting this error. Did you clone your own instance of the pipeline, then modify those parameters there?
> > > 
> > > 
> > > @bltsai-UCLA you can increase the memory for the process using [base_resource_update](https://github.com/uclahs-cds/pipeline-call-sSNV/tree/main?tab=readme-ov-file#base-resource-allocation-updaters) in the config file.
> > 
> > 
> > What version of call-sSNV are you using?
> 
> This was from metapipeline-DNA v5.2.1. I'll try again with the most updated version.

Issue is resolved with metapipeline-DNA v6.0.0.rc2
